src/test/run-pass/zero-sized-vec-deque-push.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+use std::collections::VecDeque;
+use std::iter::Iterator;
+
+#[test]
+fn test_zst_push() {
```

Comment:
In a run-pass test, you need a main function and just call this test function manually. We don't use `#[test]` here.